### PR TITLE
Update version numbers in doc

### PR DESCRIPTION
**Pull request by kimlisa**
### Purpose

Fix version `4.1.0` to `4.2.1` as I go through the `4.2` docs
And fixing minor errors I found.

### CODE REVIEW

File: docs/4.2/installation.md [hunk 1]
```diff
@@ -113,16 +113,16 @@ $ which teleport
 
 ### Windows (tsh client only)
 
-As of version v4.1.0 we have `tsh` client binary available for Windows 64-bit
+As of version v4.2.1 we have `tsh` client binary available for Windows 64-bit
```
Comment (webvictim) [inline docs/4.2/installation.md]: We’ve had the Windows client available since v3.1 or earlier I think - this isn’t one that should be updated.

**Review by webvictim — CHANGES_REQUESTED** [review]
This is definitely a good case for us adding a “latest” version URL which redirects so we don’t have to do these updates often.

**Comment by webvictim** [comment]
ok to test

File: docs/4.2/quickstart.md [hunk 2]
```diff
@@ -164,10 +164,10 @@ accessible IP.
 
 !!! warning "Warning":
     For the purposes of this quickstart we are using the
-    `--insecure-no-tls` flag which allows us to skip configuring the HTTP/TLS
```
Comment (kimlisa) [inline docs/4.2/quickstart.md]: I tested this flag out with tsh, and it didn't work, so I think it meant `--insecure` instead

File: docs/4.2/installation.md (hunk 1 above)
Comment (kimlisa) [reply docs/4.2/installation.md]: I searched through release history and found support starting from v3.0.1, so I used this version instead.

**Comment by kimlisa** [comment]
@benarent @webvictim I think... I really messed up on how I merged your guy's commits. I am not sure why your commits that are pushed to master, that I supposedly merged with my branch, is part of my changes? Or maybe this is the correct behavior?

File: docs/4.2/quickstart.md (hunk 2 above)
Comment (webvictim) [reply docs/4.2/quickstart.md]: Yeah, this causes some confusion.

For `tsh`, the flag is `--insecure` to allow insecure connections.
For `teleport`, the flag is `--insecure-no-tls` to serve the web interface via HTTP rather than HTTPS.

**Comment by webvictim** [comment]
So the reason that the commits to `master` are part of your branch is because I clicked the "Update Branch" button on Github. This basically pulls all of the commits that have been made to `master` since you started your branch and integrates them into yours. When your branch is merged back, `git` will realise that there are some commits made that are part of both branches and it will just disregard them. For that reason, it doesn't matter too much that they're present here in your branch.

In general, it's good to keep your branch up to date with `master` regularly so that you avoid having to fix any big merge conflicts when you ultimately try to merge your branch back into `master`.

One way to do this a bit more 'efficiently' is to `rebase` rather than `merge`, which works like this picture:

git-rebasing-example

Essentially it takes your branch and rewrites it so that all your commits will be made against the newest version of `master` rather than an older version.

At Gravitational we always use the "squash and rebase" option on Github when merging pull requests which tidies up the commit history anyway.

TL;DR Don't worry, you haven't done anything wrong. I'm sorry for messing with your branch. Just make sure that you run `git pull` on your local copy every now and then :)

File: docs/4.2/quickstart.md (hunk 2 above)
Comment (kimlisa) [reply docs/4.2/quickstart.md]: @webvictim Oh wow, that needs to be clarified:

> For the purposes of this quickstart we are using the `--insecure-no-tls` flag which allows us to skip configuring the HTTP/TLS

The quickstart did not use teleport `--insecure-no-tls`, so the above statement is false. Should I change the quickstart, to start teleport with `--insecure-no-tls`? https://gravitational.com/teleport/docs/quickstart/#step-2-start-teleport

Comment (webvictim) [reply docs/4.2/quickstart.md]: Yes, please do.

**Review by benarent — APPROVED** [review]

**Comment by benarent** [comment]
Looks good @kimlisa 🍾 on your first Teleport Docs PR!

File: docs/4.2/quickstart.md (hunk 2 above)
Comment (kimlisa) [reply docs/4.2/quickstart.md]: @webvictim, I run into an issue when I `teleport start --insecure-no-tls`. It seems to start. 

But I run into problems:
- I can access the web UI without using https, but I cannot log in
- logging in through the CLI `tsh --proxy=kimlisa --insecure login`, gives me this error: 

```WARNING: You are using insecure connection to SSH proxy https://kimlisa:3080
error: Get https://kimlisa:3080/v1/webapi/ping: http: server gave HTTP response to HTTPS client```

Am I doing something wrong? Or is this a bug?

Comment (webvictim) [reply docs/4.2/quickstart.md]: So I think the issue is this:

`--insecure-no-tls` is designed for when people use Teleport behind load balancers which terminate SSL/TLS themselves - it means that the load balancer can pass web requests to Teleport without needing to use HTTPS itself. There is still always the expectation that the `tsh` client will use HTTPS to communicate with Teleport, though - and that the HTTPS needed here will be provided by the load balancer rather than Teleport itself.

`--insecure` means that the `tsh` client does not validate that the SSL/TLS certificate presented by Teleport is valid i.e. it has a correct hostname, is not an expired certificate, etc. When Teleport is started without a SSL/TLS certificate being specified (using the `https_key_file` and `https_cert_file` values in the Teleport config file) then it automatically generates a self-signed certificate, but this will not pass strict validation. So for quickstarts, we use the `--insecure` parameter to disable certificate validation and to allow the `tsh` client to connect.

Comment (webvictim) [reply docs/4.2/quickstart.md]: I think for now you should probably just merge this PR. I'll take a quick pass through the quickstart docs to remove the mentions of `--insecure-no-tls` and check that it makes sense.

**Review by webvictim — APPROVED** [review]
LGTM

I think for now you should probably just merge this PR. I'll take a quick pass through the quickstart docs to remove the mentions of `--insecure-no-tls` and check that it makes sense.

**Comment by benarent** [comment]
I'm going to merge this down and deploy. ( I'll update our instructions for deploying this week )
